File: VB/WinWebSolution.Module/ListViewColumnOptionsController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports System.Collections.Generic
Imports DevExpress.ExpressApp.Editors

Namespace WinWebSolution.Module
	Public Class ListViewColumnOptionsController
		Inherits ViewController(Of ListView)
		Private Sub ListView_ModelChanged(ByVal sender As Object, ByVal e As EventArgs)
			UpdateListViewColumnOptions()
		End Sub
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			AddHandler View.ModelChanged, AddressOf ListView_ModelChanged
		End Sub
		Protected Overrides Sub OnDeactivated()
			RemoveHandler View.ModelChanged, AddressOf ListView_ModelChanged
			MyBase.OnDeactivated()
		End Sub
		Protected Overrides Sub OnViewControlsCreated()
			MyBase.OnViewControlsCreated()
			UpdateListViewColumnOptions()
		End Sub
		Protected Overridable Sub UpdateListViewColumnOptions()
			Dim columnsListEditor As ColumnsListEditor = TryCast(View.Editor, ColumnsListEditor)
			If columnsListEditor IsNot Nothing Then
				For Each columnWrapper As ColumnWrapper In columnsListEditor.Columns
					Dim options As IModelListViewColumnOptions = TryCast((CType(columnsListEditor.Model.Columns, DevExpress.ExpressApp.Model.IModelColumns)).GetNode(columnWrapper.PropertyName), IModelListViewColumnOptions)
					If options IsNot Nothing Then
						If columnWrapper.AllowSortingChange AndAlso (Not options.AllowSort) Then
							columnWrapper.AllowSortingChange = False
							columnWrapper.SortOrder = DevExpress.Data.ColumnSortOrder.None
							columnWrapper.SortIndex = -1
						End If
						If columnWrapper.AllowGroupingChange AndAlso (Not options.AllowGroup) Then
							columnWrapper.AllowGroupingChange = False
							columnWrapper.GroupIndex = -1
						End If
					End If
				Next columnWrapper
			End If
		End Sub
	End Class
End Namespace

File: CS/WinWebSolution.Module/ListViewColumnOptionsController.cs
using System;
using DevExpress.ExpressApp;
using System.Collections.Generic;
using DevExpress.ExpressApp.Editors;

namespace WinWebSolution.Module {
    public class ListViewColumnOptionsController : ViewController<ListView> {
        private void ListView_ModelChanged(object sender, EventArgs e) {
            UpdateListViewColumnOptions();
        }
        protected override void OnActivated() {
            base.OnActivated();
            View.ModelChanged += ListView_ModelChanged;
        }
        protected override void OnDeactivated() {
            View.ModelChanged -= ListView_ModelChanged;
            base.OnDeactivated();
        }
        protected override void OnViewControlsCreated() {
            base.OnViewControlsCreated();
            UpdateListViewColumnOptions();
        }
        protected virtual void UpdateListViewColumnOptions() {
            ColumnsListEditor columnsListEditor = View.Editor as ColumnsListEditor;
            if(columnsListEditor != null) {
                foreach(ColumnWrapper columnWrapper in columnsListEditor.Columns) {
                    IModelListViewColumnOptions options = columnsListEditor.Model.Columns[columnWrapper.PropertyName] as IModelListViewColumnOptions;
                    if(options != null) {
                        if(columnWrapper.AllowSortingChange && !options.AllowSort) {
                            columnWrapper.AllowSortingChange = false;
                            columnWrapper.SortOrder = DevExpress.Data.ColumnSortOrder.None;
                            columnWrapper.SortIndex = -1;
                        }
                        if(columnWrapper.AllowGroupingChange && !options.AllowGroup) {
                            columnWrapper.AllowGroupingChange = false;
                            columnWrapper.GroupIndex = -1;
                        }
                    }
                }
            }
        }
    }
}
